packages/xo-web/src/xo-app/home/index.js

```diff
@@ -481,22 +481,25 @@ export default class Home extends Component {
   }
 
   get page() {
-    return this.state.page
-  }
-  set page(activePage) {
-    this.setState({ activePage })
+    const {
+      location: { query },
+    } = this.props
+    const queryPage = +query.p
+    return !Number.isNaN(queryPage) ? queryPage : 1
   }
 
   componentWillMount() {
     this._initFilterAndSortBy(this.props)
   }
 
   componentWillReceiveProps(props) {
+    const { type } = props
+
     if (this._getFilter() !== this._getFilter(props)) {
       this._initFilterAndSortBy(props)
     }
-    if (props.type !== this.props.type) {
-      this.setState({ activePage: undefined, highlighted: undefined })
+    if (type !== this.props.type) {
+      this.setState({ highlighted: undefined })
```

Comment:
We need to change the page in this case also.

Reply:
After testing, I don't think it's necessary because we change it in `_setType` and when we click from left menu, we trigger `_setFilter` who set the page to 1.